cpp/src/barretenberg/dsl/acir_format/recursion_constraint.cpp

```diff
@@ -0,0 +1,346 @@
+#include "recursion_constraint.hpp"
+#include "barretenberg/plonk/proof_system/verification_key/verification_key.hpp"
+#include "barretenberg/stdlib/recursion/aggregation_state/aggregation_state.hpp"
+#include "barretenberg/stdlib/recursion/verifier/verifier.hpp"
+#include "barretenberg/transcript/transcript_wrappers.hpp"
+
+namespace acir_format {
+
+using namespace proof_system::plonk;
+
+static constexpr uint64_t TWO_LIMBS_BITS_IN_FIELD_SIMULATION = NUM_LIMB_BITS_IN_FIELD_SIMULATION * 2;
+static constexpr uint64_t FOUR_LIMBS_BITS_IN_FIELD_SIMULATION = NUM_LIMB_BITS_IN_FIELD_SIMULATION * 4;
```

Comment:
We should document what these constants are for

Reply:
Documented let me know if the description makes sense

Reply:
lgtm 👍